nixos/modules/services/monitoring/beszel-agent.nix

```diff
@@ -0,0 +1,107 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.beszel.agent;
+in
+{
+  meta.maintainers = with lib.maintainers; [ BonusPlay ];
+
+  options.services.beszel.agent = {
+    enable = lib.mkEnableOption "beszel agent";
+
+    package = lib.mkPackageOption pkgs "beszel" { };
+
+    user = lib.mkOption {
```

Comment:
Is there any use case to running beszel under a different user?
Otherwise I would suggest to remove that option.

Reply:
no, but it's useful for setting secret ownership (like environment variables)

Reply:
You don't need an extra user for that, only root needs to be able to read a file for systemd's EnvironmentFile.
DynamicUser=true doesn't do anything if there is already a user with that name. 
We should use DynamicUser over an extra user if there is no good reason not to.

Reply:
I think you're right in this case. It helps when something is config file (loaded during runtime) and not env file. Applied suggestion :+1:.